http_parser.h

```diff
@@ -302,7 +301,8 @@ struct http_parser {
   unsigned int state : 7;        /* enum state from http_parser.c */
   unsigned int header_state : 7; /* enum header_state from http_parser.c */
   unsigned int index : 5;        /* index into current matcher */
-  unsigned int extra_flags : 2;
+  unsigned int uses_transfer_encoding : 1; /* Transfer-Encoding header is present */
+  unsigned int allow_nonrfc_clients : 1;  /* allow headers with both `Content-Length` and `Transfer-Encoding: chunked` set */
```

Comment:
I think the member name could be a bit better, it seems like it could easily be confused with `lenient_http_headers`. Here's a humble idea for something not too long but more descriptive: `allow_length_with_encoding`

Reply:
np, renamed!

Reply:
... thinking a bit more, there are other Transfer-Encoding that are not a problem with Content-Length, maybe a better name is `allow_length_chunked` or `allow_chunked_length`

Reply:
allow_chunked_length then, updated

Reply:
looks good to me, thanks